### ISSUE TITLE
Warning: `controlId` is ignored on `<FormControl>` when `id` is specified.

### ISSUE THREAD

**Issue by twschiller** [pixiebrix/pixiebrix-extension#1429]
Our Form framework is generating this warning, which makes looking at error counts in the console hard

**Comment by BALEHOK** [comment]
It looks like the warning is generated by the Bootstrap FormControl when it is used within a custom widget.
For instance, the output field:
<img width="872" alt="Screen Shot 2021-09-24 at 1 18 00 PM" src="https://user-images.githubusercontent.com/3116723/134666173-0ed708f2-29f3-4ab2-81dd-7925df759e67.png">

The implementation:
```
<Col xl>
          <ConnectedFieldTemplate
            name={`${blockFieldName}.outputKey`}
            label={<PopoverOutputLabel description={outputDescription} />}
            disabled={isOutputDisabled}
            as={OutputKeyWidget}
          />
</Col>
```

and the OutputKeyWidget (https://github.com/pixiebrix/pixiebrix-extension/blob/6fe621d6e524739e829d9198cb796f05b043df25/src/devTools/editor/tabs/editTab/editorNodeConfigPanel/EditorNodeConfigPanel.tsx#L33):
```
const OutputKeyWidget: CustomFieldWidget = (props: FieldProps) => (
  <InputGroup>
    <InputGroup.Prepend>
      <InputGroup.Text>@</InputGroup.Text>
    </InputGroup.Prepend>
    <Form.Control {...props} />
  </InputGroup>
);
```

Problem here is that FieldTemplate uses `Form.Control` to render the widget and then the widget renders `Form.Control` again. As a result we have a `Form.Control` rendered inside another `Form.Control`. It seems that the internal `Form.Control` got `id` property set twice.

A quick fix is to remove `id` from props:
```
const OutputKeyWidget: CustomFieldWidget = (props: FieldProps) => {
  const { id, ...restProps } = props;
  return (
    <InputGroup>
      <InputGroup.Prepend>
        <InputGroup.Text>@</InputGroup.Text>
      </InputGroup.Prepend>
      <Form.Control {...restProps} />
    </InputGroup>
  );
};
```

**Comment by BALEHOK** [comment]
Ideally, the widgets need to be placed in a common folder. Another improvement is not to render `From.Control` in the `FieldTemplate` when a widget specified:
```
<BootstrapForm.Group controlId={name} className={styles.verticalFormGroup}>
      {label && (
        <BootstrapForm.Label className={styles.verticalFormLabel}>
          {label}
        </BootstrapForm.Label>
      )}
     {As
     ? (<As name={name}
            isInvalid={isInvalid}
            value={nonUndefinedValue}
            {...restFieldProps} />)
      : (<BootstrapForm.Control
            name={name}
            isInvalid={isInvalid}
            value={nonUndefinedValue}
            {...restFieldProps}
          >
            {children}
          </BootstrapForm.Control>
      )}
      {description && (
        <BootstrapForm.Text className="text-muted">
          {description}
        </BootstrapForm.Text>
      )}
      {isInvalid && (
        <div className={styles.invalidMessage}>{getErrorMessage(error)}</div>
      )}
    </BootstrapForm.Group>
```

**Comment by BALEHOK** [comment]
Solved by PR #1452
